src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1770,7 +1770,7 @@ public override void VisitMarkupMinimizedTagHelperDirectiveAttribute(MarkupMinim
                             }
                             else
                             {
-                                attributeNode = new TagHelperPropertyIntermediateNode()
+                                attributeNode = new TagHelperDirectiveAttributeIntermediateNode()
                                 {
                                     AttributeName = attributeName,
```

Comment:
Once converted into a directive attribute do we ever need the `@` again? Wondering if we can just drop that piece from the IR entirely at this point.

Reply:
Well, doing that would make more sense if the `BoundAttributeDescriptor` for the directive attribute doesn't contains the `@`. But it does, and it doesn't feel right for the `AttributeName` to differ from what the bound attribute descriptor says it is. I'm open to arguments otherwise though.

Reply:
> it doesn't feel right for the `AttributeName` to differ from what the bound attribute descriptor says

Eh, it kind of does already with indexer attributes. I'd change it

Reply:
> Eh, it kind of does already with indexer attributes.

Not sure what you mean. For an attribute that looks like `@bind-value=...`, the attribute name will be `@bind-value` and the indexer name prefix will be `@bind-` and the bound attribute name will be `@bind-...`. This makes sense to me.